examples/loki/nurf_data.py

```diff
@@ -102,120 +102,205 @@ def nurf_file_creator(loki_file, path_to_loki_file, data):
 
     # open the file and append
     with h5py.File(loki_file, 'a') as hf:
+                
+        #comment on names
+        #UV/FL_Background is the dark
+        #UV/FL_Intensity0 is the reference
+        #UV/FL_Spectra is the sample
         
+        # image_key: number of frames (nFrames) given indirectly as part of the shape of the arrays 
+        # TODO: keep in mind what happens if multiple dark or reference frames are taken
+        
+        # remove axis=2 of length one from this array
+        data['UV_spectra']=np.squeeze(data['UV_spectra'], axis=2)  #this removes the third axis in this array, TODO: needs later to be verified with real data from hardware
+        
+        # assemble all spectra in one variable
+        uv_all_data=np.row_stack((data['UV_spectra'],data['UV_background'], data['UV_intensity0']))
+      
+        
+        # assemble image_key 
+        # #TODO: needs later to be verified with real data from hardware
+        uv_nb_spectra=np.shape(data['UV_spectra'])[0]
+        uv_ik_spectra=np.zeros((1,uv_nb_spectra))  #interperation here: 0 for sample (in comparison to projections)
+      
+        # find out how many nFrames each item (sample, dark, reference) has
+        if data['UV_background'].ndim==1:
+            uv_nb_darks=1
+        else: 
+            uv_nb_darks=np.shape(data['UV_background'])[1]  #TODO: needs to be verified with real data from Judith's setup
```

Comment:
Shouldn't this be `[0]`? I thought `[1]` would give you the number of wavelength points.

Reply:
No, this is here correct. The person who put in the hdf5 file I am using data['UV_background'], has it as one column with many rows. Wavelength points are for this real data in axis 0, not 1.
Opposite to data['UV_spectra], where there are two rows for the 2 spectra and in dimension one the wavelengths.
If we use our own setup in the lab, we can fix this.

Reply:
Then how does the `np.row_stack` above work?

Reply:
It does not work anymore, I replaced it with `np.hstack`.

Reply:
Are you sure that is correct? Please have a look at https://stackoverflow.com/a/65177565/1458281.

Reply:
I check this later, I have to go now to a longer meeting.